src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationOptions.cs

```diff
@@ -58,7 +58,7 @@ public OpenIdConnectAuthenticationOptions(string authenticationScheme)
             BackchannelTimeout = TimeSpan.FromMinutes(1);
             Caption = OpenIdConnectAuthenticationDefaults.Caption;
             GetClaimsFromUserInfoEndpoint = false;
-            ProtocolValidator = new OpenIdConnectProtocolValidator();
+            ProtocolValidator = new OpenIdConnectProtocolValidator() { RequireState = false };
```

Comment:
We need to use the same strategy as the 'nonce', we include it as part of the nonce. Validation requires that the state returned in the Response message is exactly the same as sent in the Request message.